### PR TITLE
WIP: use dotnet-xunit to run *.Tests.fsproj.  Allows for running on mono.

### LINKED ISSUE #190: Running tests on mono-net461
`dotnet test` is broken on mono (https://github.com/Microsoft/vstest/issues/1318) but if we use `dotnet-xunit` I added support for mono a while ago.  We had this discussion previously in https://github.com/giraffe-fsharp/Giraffe/pull/58 but seems like it's not in beta anymore so this may be worth considering.  Thoughts?

### PR DESCRIPTION
**Pull request by TheAngryByrd**
Closes #190. This is the 1st pass at using `dotnet-xunit` to run the tests for Giraffe.Tests.fsproj.  I'm not a powershell expert so please criticize.   

Due to some weird behavior in dotnet's cli tooling (https://github.com/xunit/xunit/issues/1573 and https://github.com/dotnet/cli/issues/7901#issuecomment-352009367) we have to pass the `-fxversion` to `dotnet-xunit`.  

```
dotnet xunit -fxversion 2.0.3
Detecting target frameworks in Giraffe.Tests.fsproj...
Building for framework net461...
  Giraffe -> /Users/username/Documents/GitHub/Giraffe/src/Giraffe/bin/Debug/net461/Giraffe.dll
  Giraffe.Tests -> /Users/username/Documents/GitHub/Giraffe/tests/Giraffe.Tests/bin/Debug/net461/Giraffe.Tests.dll
Running desktop CLR tests for framework net461...
xUnit.net Console Runner (64-bit Desktop .NET 4.0.30319.42000)
  Discovering: Giraffe.Tests
  Discovered:  Giraffe.Tests
  Starting:    Giraffe.Tests
  Finished:    Giraffe.Tests
=== TEST EXECUTION SUMMARY ===
   Giraffe.Tests  Total: 139, Errors: 0, Failed: 0, Skipped: 0, Time: 2.171s
Building for framework netcoreapp2.0...
  Giraffe -> /Users/username/Documents/GitHub/Giraffe/src/Giraffe/bin/Debug/netstandard2.0/Giraffe.dll
  Giraffe.Tests -> /Users/username/Documents/GitHub/Giraffe/tests/Giraffe.Tests/bin/Debug/netcoreapp2.0/Giraffe.Tests.dll
Running .NET Core 2.0.3 tests for framework netcoreapp2.0...
xUnit.net Console Runner (64-bit .NET Core 4.6.0.0)
  Discovering: Giraffe.Tests
  Discovered:  Giraffe.Tests
  Starting:    Giraffe.Tests
  Finished:    Giraffe.Tests
=== TEST EXECUTION SUMMARY ===
   Giraffe.Tests  Total: 139, Errors: 0, Failed: 0, Skipped: 0, Time: 1.137s
```

```
$ mono -V
Mono JIT compiler version 5.4.1.6 (2017-06/1f4613aa1ac Wed Oct 18 09:31:57 EDT 2017)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug
	LLVM:          yes(3.6.0svn-mono-master/8b1520c8aae)
	GC:            sgen (concurrent by default)
```

### CODE REVIEW

File: tests/Giraffe.Tests/Giraffe.Tests.fsproj [hunk 1]
```diff
@@ -18,6 +18,9 @@
     <PackageReference Include="NSubstitute" Version="3.1.*" />
     <PackageReference Include="FsCheck" Version="3.0.0-*" />
     <PackageReference Include="FsCheck.Xunit" Version="3.0.0-*" />
+    <DotNetCliToolReference Include="dotnet-xunit" Version="2.3.1" />
+    <DotNetCliToolReference Include="Microsoft.DotNet.Watcher.Tools" Version="2.0.0" />
```
Comment (dustinmoris) [inline tests/Giraffe.Tests/Giraffe.Tests.fsproj]: What is the Watcher used for?

**Comment by dustinmoris** [comment]
Hey did you want to apply the same to the sample project tests or shall I merge it as it is?

**Comment by TheAngryByrd** [comment]
Yeah I was wondering if you'd want me to apply it it all. I will of this seems appropriate.

File: tests/Giraffe.Tests/Giraffe.Tests.fsproj (hunk 1 above)
Comment (TheAngryByrd) [reply tests/Giraffe.Tests/Giraffe.Tests.fsproj]: I usually put this on all applications. This allows me.to write `dotnet watch command` and have it rebuild and rerun/retest whenever I save a file. 

Can remove.it needed.

**Pull request merged**